### ISSUE TITLE
OSX Jenkins is missing `enchant`

### ISSUE THREAD

**Issue by betatim** [dib-lab/khmer#1417]
The OSX build is failing because of a missing C library `enchant`. @standage or @ctb who do I talk to to get it installed? I couldn't find the script used to configure the environment for the CI builds. The following console output is taken from #1416 and you see the same in others.

```
Downloading/unpacking pyenchant
  Running setup.py (path:/Users/jenkins/workspace/khmer-pullrequest/label/osx/.env/build/pyenchant/setup.py) egg_info for package pyenchant
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/Users/jenkins/workspace/khmer-pullrequest/label/osx/.env/build/pyenchant/setup.py", line 210, in <module>
        import enchant
      File "enchant/__init__.py", line 92, in <module>
        from enchant import _enchant as _e
      File "enchant/_enchant.py", line 145, in <module>
        raise ImportError(msg)
    ImportError: The 'enchant' C library was not found. Please install it via your OS package manager, or use a pre-built binary wheel from PyPI.
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/Users/jenkins/workspace/khmer-pullrequest/label/osx/.env/build/pyenchant/setup.py", line 210, in <module>

    import enchant

  File "enchant/__init__.py", line 92, in <module>

    from enchant import _enchant as _e

  File "enchant/_enchant.py", line 145, in <module>

    raise ImportError(msg)

ImportError: The 'enchant' C library was not found. Please install it via your OS package manager, or use a pre-built binary wheel from PyPI.
```

**Comment by standage** [comment]
I am about 6 inches from the iMac that runs Jenkins builds, but I don't have credentials to access the machine. I'll see if Michael wrote these down anywhere.

**Comment by mr-c** [comment]
The `imac.oxli.org` hostname isn't working anymore so I can't log in remotely to fix this :-/

**Comment by standage** [comment]
Power's on, but not connected to ethernet cable. So unless it was using wireless that's probably the problem.

**Comment by camillescott** [comment]
Also running into this. Does it need to be given a static IP?

**Comment by mr-c** [comment]
It had a static IP, yes. Maybe it expired when I left UCDavis?

Pe 16 aug. 2016 06:22, "Camille Scott" <notifications@github.com> a scris:

**Comment by betatim** [comment]
Given #1420 do you want to keep investigating this?

I think the solution should be `brew install enchant` (or equivalent). So if you can connect and fix it this easily that would be great whiel waiting for #1420

**Comment by standage** [comment]
I'd be happy to work on this if somebody with admin privileges on the machine could pass the baton. I currently don't have an account on the machine.

**Comment by standage** [comment]
Etherned cable connected now.

**Comment by mr-c** [comment]
Since the imac is still connected to Jenkins over the wifi I was able to get its IP address by going through Jenkins

http://ci.oxli.org/computer/imac/script

`println "ifconfig -a".execute().text`

@standage I've created a user account for you there following the directions at http://apple.stackexchange.com/questions/82472/what-steps-are-needed-to-create-a-new-user-from-the-command-line
One change, I used `dscl . -list /Users UniqueID` to discover what user IDs are available. The password is the one from the other day. According to [this source](http://apple.stackexchange.com/questions/226073/how-do-i-create-user-accounts-from-the-terminal-in-mac-os-x-10-11) you won't be able to see yourself with Admin privilages until after the next reboot, but you do indeed have them.

**Comment by ctb** [comment]
(thanks @mr-c!)

**Comment by standage** [comment]
Awesome @mr-c! I was able to log in this morning, change my password, reboot, and confirm admin privileges. Thanks for your help!

@betatim I've done some configuration and updates.

- [Adjusted permissions so that any iMac user with admin priv can manage Hombrew](http://apple.stackexchange.com/a/45003/1870)
- updated Homebrew with `brew update && brew upgrade`
- installed Enchant with `brew install enchant`
- installed Enchant Python bindings (and upgraded pip) with `sudo pip install --upgrade pip pyenchant`

I ran `python -c 'import enchant'` as user `jenkins` and confirmed it is installed correctly. Let's confirm this with the next build, but hopefully this issue should be solved.
